tests/integration.spec.coffee

```diff
@@ -1829,6 +1829,26 @@ describe 'SDK Integration Tests', ->
 						promise = resin.models.os.download('foo-bar-baz')
 						m.chai.expect(promise).to.be.rejectedWith('No such device type')
 
+			describe 'resin.models.os.getConfig()', ->
+				before ->
+					resin.models.application.create('TestApp', 'raspberry-pi').then (application) =>
+						@application = application
+
+				it 'should be able to get an application config by id', ->
+					promise = resin.models.os.getConfig(@application.app_id)
+					m.chai.expect(promise).to.eventually.have.property('applicationId')
+					m.chai.expect(promise).to.eventually.have.property('apiKey')
+					m.chai.expect(promise).to.eventually.have.property('userId')
+					m.chai.expect(promise).to.eventually.have.property('username')
+					m.chai.expect(promise).to.eventually.have.property('deviceType')
+					m.chai.expect(promise).to.eventually.have.property('files')
+					m.chai.expect(promise).to.eventually.have.property('apiEndpoint')
+					m.chai.expect(promise).to.eventually.have.property('registryEndpoint')
+					m.chai.expect(promise).to.eventually.have.property('vpnEndpoint')
+					m.chai.expect(promise).to.eventually.have.property('pubnubSubscribeKey')
+					m.chai.expect(promise).to.eventually.have.property('pubnubPublishKey')
+					m.chai.expect(promise).to.eventually.have.property('listenPort')
```

Comment:
We should add another test to check you get a sensible error if the application id doesn't exist.